pages/[id]/intro.js

```diff
@@ -0,0 +1,247 @@
+import styled from 'styled-components';
+import { useState, useEffect } from 'react';
+import Page from '../../components/page';
+import Card, {
+  CardHeader,
+  CardTitle,
+  CardContent,
+  CardButtonContainer,
+} from '../../components/card';
+import { COLOR, EDIT } from '../../constants';
+import Button from '../../components/button';
+import TextBox from '../../components/textbox';
+import {
+  updateHackathonField,
+  getHackathon,
+  getHackathonPaths,
+  getHackathons,
+} from '../../utility/firebase';
+
+const HeaderText = styled.p`
+  border: none;
+  overflow: hidden;
+  background-color: ${COLOR.BACKGROUND};
+  outline: none;
+  font-size: 24px;
+  padding-bottom: 13px;
+  color: ${COLOR.TEXT};
+  font-weight: 600;
+`;
+
+const ContentText = styled.p`
+  border: none;
+  overflow: hidden;
+  background-color: ${COLOR.BACKGROUND};
+  outline: none;
+  color: ${COLOR.TEXT};
+`;
+
+const StyledCancel = styled.button`
+  font-size: 16px;
+  cursor: pointer;
+  border-bottom: 2px solid ${COLOR.BLACK};
+  margin-left: 675px;
+  margin-right: 40px;
+  border: none;
+  outline: none;
+  background-color: ${COLOR.BACKGROUND};
+`;
+
+const StyledHeader = styled.p`
+  padding-bottom: 13px;
+  padding-top: 17px;
+  margin: 0px;
+  background-color: ${COLOR.BACKGROUND};
+`;
+
+export default ({ id, hackathons }) => {
+  const [isEditingObj, setIsEditingObj] = useState({});
+  const [websiteData, setWebsiteData] = useState({});
+  const [editingData, setEditingData] = useState({});
+  const currUserName = 'SOME_HARD_CODED_USER';
+  // TODO need functionality to display data based on what event is currently being viewed
+
+  // TODO need to substitute the value at .doc(ID) with current website
+  const getHackathonData = async () => {
+    const doc = await getHackathon(id);
+    const data = doc.data().WebsiteData;
+    setWebsiteData(data);
+
+    let editingDataObj = {};
+    let isEditing = {};
+    Object.keys(data).forEach((type) => {
+      isEditing = {
+        ...isEditingObj,
+        [type]: false,
+      };
+      editingDataObj = {
+        ...editingDataObj,
+        [type]: {
+          header: data[type].header,
+          content: data[type].content,
+        },
+      };
+    });
+    setIsEditingObj(isEditing);
+    setEditingData(editingDataObj);
+  };
+
+  useEffect(() => {
+    getHackathonData();
+  }, []);
+
+  const handleEdit = (type) => {
+    setIsEditingObj({
+      ...isEditingObj,
+      [type]: !isEditingObj[type],
+    });
+  };
+
+  const handleCancel = (type) => {
+    setIsEditingObj({
+      ...isEditingObj,
+      [type]: false,
+    });
```

Comment:
If I make an edit and press cancel, what happens when I click edit again? Will my edits still be there?

Reply:
yah, this was intentional and discussed with giullio